common/persistence/dataManagerInterfaces.go

```diff
@@ -1435,6 +1435,7 @@ type (
 		FailoverEndTime             *int64
 		LastUpdatedTime             int64
 		NotificationVersion         int64
+		IsolationGroups             []types.IsolationGroupPartition
```

Comment:
When do we use `types.IsolationGroupConfiguration` and when do we use `[]types.IsolationGroupPartition`?

Reply:
BTW, this new field doesn't seem to be used anywhere.

Reply:
that's a relic I thnk, sorry, let me remove it

Reply:
To answer your question more directly: 

Internally, *everywhere* internally we use types.IsolationGroupConfiguration and  only in IDL facing the user or DB we flatten this to `[]types.IsolationGroupPartition` as it's less confusing.